hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/states/endpoint/HeartbeatEndpointTask.java

```diff
@@ -83,21 +87,50 @@
   private StateContext context;
   private int maxContainerActionsPerHB;
   private int maxPipelineActionsPerHB;
+  private HDDSLayoutVersionManager layoutVersionManager;
 
   /**
    * Constructs a SCM heart beat.
    *
+   * @param rpcEndpoint rpc Endpoint
    * @param conf Config.
+   * @param context State context
    */
   public HeartbeatEndpointTask(EndpointStateMachine rpcEndpoint,
```

Comment:
We can reuse the lines of constructor method, like this
public HeartbeatEndpointTask(EndpointStateMachine rpcEndpoint,
      ConfigurationSource conf, StateContext context) {
   this(rpcEndpoint, conf, context, context.getParent().getDataNodeVersionManager())
  }
  
  public HeartbeatEndpointTask(EndpointStateMachine rpcEndpoint,
                               ConfigurationSource conf, StateContext context,
                               HDDSLayoutVersionManager versionManager) {
    ...
    layoutVersionManager = versionManager;
  }

Reply:
yup